vcd/resource_vcd_independent_disk.go

```diff
@@ -92,10 +151,19 @@ func resourceVcdIndependentDiskCreate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("disk with such name already exist : %s", diskName)
 	}
 
-	diskCreateParams := &types.DiskCreateParams{Disk: &types.Disk{
-		Name: diskName,
-		Size: int64(d.Get("size").(float64) * 1024 * 1024),
-	}}
+	var diskCreateParams *types.DiskCreateParams
+	if sizeProvided {
+		diskCreateParams = &types.DiskCreateParams{Disk: &types.Disk{
+			Name: diskName,
+			Size: int64(size.(float64) * 1024 * 1024),
+		}}
+	}
+	if sizeInBytesProvided {
+		diskCreateParams = &types.DiskCreateParams{Disk: &types.Disk{
+			Name: diskName,
+			Size: int64(sizeInBytes.(int)),
+		}}
+	}
```

Comment:
Would like to ask additional eyes of @dataclouder and @Didainius to review the deprecation of `size` in favour of `size_in_bytes`. We need to do this, but want to have all of us to think through of possible corner cases of backward-compatibility. There will be more deprecations like this going forward.

Reply:
Looks like the "read" method is missing `_ = d.Set("size", disk.Disk.Size*....)`
I know it is being deprecated, but I see this field it is still provided in data source. The data source will have an invalid value for it because "read" does not set it. 

IMHO we should omit field `size` from data source at all and do not read it. That way whoever has it, the value will stay in place, but point towards `size_in_bytes`

Reply:
I don't like this deprecation, as it leaves users with a poor reference in the UI, where you can only define a disk size in either MB or GB.